### DISCUSSION TITLE
コマンド実行の設計に関する議論

### DISCUSSION THREAD

**Discussion by ntk221** [ntk221/minishell#49] [General]
<img width="322" alt="スクリーンショット 2022-12-27 15 46 24" src="https://user-images.githubusercontent.com/92190938/209623852-afc653ea-a740-4380-81c1-faffd8065679.png">

**Comment by ntk221** [comment]
今現在，`ft_echo`がこの問題に少し陥っているように感じる

**Comment by nekotonatsu** [comment]
コマンドがある程度限定されているなら
事前にコマンドごとの優先度を決めておく
⇒　入力値をいったんどこかの配列に入れる
⇒　コマンドがどこにあるか確認
⇒　優先度順位から順番に取っていく
という事が出来ればいいのでは、と感じている

**Reply by ntk221** [reply]
確かにその作戦でもできそうかも！

しかし、コマンドラインにおける、”コマンドの優先度”が個々の具体的なコマンドラインと無関係に事前に決定できるようなものなのか？という疑問があります。

結局あるコマンドラインについて、実行するコマンドの順番を決定するには、一度コマンドライン上を走査して、優先度を決定するという作業が必要になるのでは、ということです。ペアプロ時に具体的に議論しましょう！
